code/game/objects/structures/skele_stand.dm

```diff
@@ -0,0 +1,48 @@
+/obj/structure/skele_stand
+	name = "hanging skeleton model"
+	anchored = 0
+	density = 1
+	icon = 'icons/obj/surgery.dmi'
+	icon_state = "hangskele"
+	desc = "It's an anatomical model of a human skeletal system made of plaster."
+	var/last_rattle_time = 0
+	var/rattle_cooldown = 2 SECONDS// let's not get our rattling out of hand
+
+/obj/structure/skele_stand/New()
+	..()
+	gender = pick(MALE, FEMALE)
+
+/obj/structure/skele_stand/proc/rattle_bones(mob/user, atom/thingy)
+	if(last_rattle_time + rattle_cooldown <= world.time)
+		if(user)
+			visible_message("\The [user] pushes on [src][thingy?" with \the [thingy]":""], giving the bones a good rattle.")
+		else
+			visible_message("\The [src] rattles on \his stand upon being hit by [thingy?"\the [thingy]":"something"].")
```

Comment:
You can extend the check to do `[thingy ? "upon being hit by \the [thingy]" : ""]`, that way if thingy is null, you don't get the hit part of the message.